cmd/controller/main.go

```diff
@@ -264,6 +270,7 @@ func parseEnvFlags() config {
 	viper.SetDefault(logDirFlag, "")
 	viper.SetDefault(logLevelFlag, "Info")
 	viper.SetDefault(logSizeLimitMBFlag, 10000) // 10 GB, will be split into 100 MB chunks
+	viper.SetDefault(cloudProductFlag, "")
```

Comment:
Non blocking suggestion:

```suggestion
	viper.SetDefault(cloudProductFlag, "auto")
```

Only change if you like it.

Reply:
If I go that way, do you think the empty string should be invalid, then? (i.e. we rewrite --cloud-product to be set to something valid, but it defaults to `auto`)

Reply:
That would be my thought, but also what happens if I enter "blueberry" 😄  does it default to auto detect?

Maybe if the value passed in is not known, then it throws a warning, and does an autodetect?

Reply:
I like this approach, will get a version up soon.

Reply:
Awesome. Otherwise, LGTM!

Reply:
Oh, I somehow missed this comment

> Maybe if the value passed in is not known, then it throws a warning, and does an autodetect?

The version I uploaded throws an error if it doesn't know the product (and on "") - I think it's reasonable to do this on CLI flags, since overriding them is "explicit intent".